packages/common/rollup-configs.js

```diff
@@ -100,5 +100,8 @@ export const dtsLibConfig = (pkg, inputFile) => ({
       summaryOnly: true,
       root: rootDir,
     }),
+    removeFiles({
+      files: ['dist/types'],
```

Comment:
`dist/types` directory contains a `.d.ts` file per each module (following the package source file structure), which gets bundled into a single `dist/index.d.ts` file. To avoid output artifact clutter, we remove this directory once we're done processing its content.

Reply:
@vojtechszocs a few things we may want to consider:
1. Should we log anything during the build process about which files are being removed in case we run into a bug where some unintended files are being removed? Not sure if rollup does this for us with a `debug` flag or similar.
2. What is the risk of leaving the duplicate/nested `.d.ts` files? Is there a risk those will be used instead of the `dist/index.d.ts`?

Reply:
When we build a library, we use two Rollup configs:
- `tsLibConfig` which performs TS to JS transpilation and bundling into `dist/index.js`
  - as a side effect of `"declaration": true` setting, this also generates the `dist/types` content
- `dtsLibConfig` which performs the bundling of `dist/index.d.ts` from ^^ `dist/types` content
  - once this is complete, the `dist/types` content is technically no longer needed

Maybe I'm over-engineering this. I'll remove this code, it may look too confusing.

Reply:
> What is the risk of leaving the duplicate/nested `.d.ts` files? Is there a risk those will be used instead of the `dist/index.d.ts`?

There's no risk, files that we publish to npmjs are https://docs.npmjs.com/cli/v8/commands/npm-publish#files-included-in-package basically `package.json`, `README.md` and files specified in `files` array in `package.json`.

We should also include a per-package `LICENSE` file but that's a follow-up improvement.